clicommand/agent_start.go

```diff
@@ -516,10 +516,11 @@ var AgentStartCommand = cli.Command{
 			Value:  1,
 			EnvVar: "BUILDKITE_AGENT_SPAWN",
 		},
-		cli.BoolFlag{
+		cli.GenericFlag{
 			Name:   "spawn-with-priority",
-			Usage:  "Assign priorities to every spawned agent (when using --spawn) equal to the agent's index",
+			Usage:  "Assign priorities to every spawned agent (when using --spawn) corresponding to the agent's index. This can be passed with or without a value. By default (without this flag, or setting it to false) each spawned agent's priority will be equal to --priority. When passed on its own, each spawned agent's priority will equal the agent's index. When passed with 'inverse', each spawned agent's priority will be assigned in reverse (counting down from --priority).",
```

Comment:
I know that i'm two-hard-problemsing us here but...

Given that we have two roughly opposite options here, it seems to me that the default behaviour should have a name, even if it's only in the docs. It seems a bit strange to me for the options to be `inverse` and.... nothing.

The numismatist in me suggests `obverse` and `reverse`, but that's probably getting a bit wanky.

`upward` and `downward`?

Reply:
tbh I'm still of the opinion that we should be changing the default behaviour, as there's no need for the existing behaviour – inverse fills the same requirements, and explaining it in documentation will be confusing

Reply:
And `inverse` accounts for `--priority` where the default doesn't, so the behaviours are no longer the inverse of each other

Reply:
`ascending` (default) versus `descending`?

Reply:
I've changed it to `ascending` / `descending`, with `*-from-priority` variations.

I'm open to being told it's too many options. 😅

I'm not certain there's no use-case for `ascending` - suppose someone wants to ensure jobs run on agents running on energy-efficient hosts first, before scaling up to less efficient hosts.

`*-from-priority` makes sense to me. Give me a lever, I'll try to pull it.

Maybe I'm too attached to preserving the existing behaviour, in order to avoid user surprises?

Reply:
I reckon just have `ascending` and `descending` and always base it on `priority`. Someone might be using a combo, but it seems unlikely, and if they do they can remove (or change) their priority. (If this is you — please remove or change your configured `priority` when combined with `spawn-with-priority` :pray:)